tests/functional/syntax/modules/test_initializers.py

```diff
@@ -158,6 +158,38 @@ def foo():
     assert compile_code(main, input_bundle=input_bundle) is not None
 
 
+def test_initializer_list_module_mismatch(make_input_bundle):
+    lib1 = """
+counter: uint256
+    """
+    lib2 = """
+something: uint256
+    """
+    lib3 = """
+import lib1
+
+uses: lib1
+
+@internal
+def foo():
+    lib1.counter += 1
+    """
+    main = """
+import lib1
+import lib2
+import lib3
+
+initializes: lib1
+initializes: lib3[lib1 := lib2]  # typo -- should be [lib1 := lib2]
```

Comment:
they're both the same though?

Reply:
oh, bad comment, nice catch

Reply:
https://github.com/vyperlang/vyper/pull/3729/commits/ca77cccf16bd695722af4c990022461dde7e0ad4